### PR TITLE
[semver] parse `^` & `~` expressions correctly

### PR DESCRIPTION
**Pull request by alexlamsl**

### CODE REVIEW

File: src/install/semver.zig [hunk 1]
```diff
@@ -1106,70 +1130,67 @@ pub const Range = struct {
...
-                var rhs = Version{};
-                rhs.major = version.major;
-                rhs.minor = version.minor;
-
-                // rhs.raw = version.raw;
-                // lhs.raw = version.raw;
-
+                const lhs = Version{
+                    .major = version.major,
+                    .minor = version.minor + 1,
```
Comment (Jarred-Sumner) [inline src/install/semver.zig]: pedantic:

```suggestion
                    .minor = version.minor +| 1,
```

File: src/install/semver.zig [hunk 2]
```diff
@@ -1418,113 +1439,125 @@ pub const Query = struct {
...
-                        },
-                    };
+                        };
+                        if (version.hasMinor()) {
+                            range.left.version.minor = version.minor;
+                            if (version.hasPatch()) {
+                                range.left.version.patch = version.patch;
+                                if (version.major == 0) {
+                                    if (version.minor == 0) {
+                                        range.right.version.patch = version.patch + 1;
```
Comment (Jarred-Sumner) [inline src/install/semver.zig]:
```suggestion
                                        range.right.version.patch = version.patch +| 1;
```

Comment (alexlamsl) [reply src/install/semver.zig:1460]: is the extra `.` intentional?

Comment (alexlamsl) [reply src/install/semver.zig]: Done in https://github.com/oven-sh/bun/pull/1854/commits/f0e686aa6b42639a78f758b9a9c76b6d046d1d85

File: src/install/semver.zig (hunk 1 above)
Comment (alexlamsl) [reply src/install/semver.zig]: Done in https://github.com/oven-sh/bun/pull/1854/commits/f0e686aa6b42639a78f758b9a9c76b6d046d1d85

**Comment by Jarred-Sumner** [comment]
yay

**Pull request merged**